drv/ksz8463/src/registers.rs

```diff
@@ -1073,33 +1074,52 @@ pub enum Register {
 #[allow(non_snake_case)]
 impl Register {
     #[inline(always)]
-    pub fn PxPHYCTRL(i: u8) -> Self {
-        Self::select(i, Self::P1PHYCTRL, Self::P2PHYCTRL)
+    pub fn PxPHYCTRL(i: KszPhyPort) -> Self {
+        Self::select2(i, Self::P1PHYCTRL, Self::P2PHYCTRL)
     }
     #[inline(always)]
-    pub fn PxMBSR(i: u8) -> Self {
-        Self::select(i, Self::P1MBSR, Self::P2MBSR)
+    pub fn PxMBSR(i: KszPhyPort) -> Self {
+        Self::select2(i, Self::P1MBSR, Self::P2MBSR)
     }
     #[inline(always)]
-    pub fn PxMBCR(i: u8) -> Self {
-        Self::select(i, Self::P1MBCR, Self::P2MBCR)
+    pub fn PxMBCR(i: KszPhyPort) -> Self {
+        Self::select2(i, Self::P1MBCR, Self::P2MBCR)
     }
     #[inline(always)]
-    pub fn PxCR1(i: u8) -> Self {
-        Self::select(i, Self::P1CR1, Self::P2CR1)
+    pub fn PxCR1(i: KszPort) -> Self {
+        Self::select3(i, Self::P1CR1, Self::P2CR1, Self::P3CR1)
     }
     #[inline(always)]
-    pub fn PxCR2(i: u8) -> Self {
-        Self::select(i, Self::P1CR2, Self::P2CR2)
+    pub fn PxCR2(i: KszPort) -> Self {
+        Self::select3(i, Self::P1CR2, Self::P2CR2, Self::P3CR2)
+    }
+
+    #[inline(always)]
+    pub fn PxVIDCR(i: KszPort) -> Self {
+        Self::select3(i, Self::P1VIDCR, Self::P2VIDCR, Self::P3VIDCR)
     }
 
     // Helper function to dispatch between two registers
     #[inline(always)]
-    fn select(i: u8, r1: Register, r2: Register) -> Register {
+    fn select2(i: KszPhyPort, r1: Register, r2: Register) -> Register {
+        match i {
+            KszPhyPort::One => r1,
+            KszPhyPort::Two => r2,
+        }
+    }
+
+    // Helper function to dispatch between three registers
+    #[inline(always)]
+    fn select3(
```

Comment:
nit, take it or leave it: now that we have enums for `KszPort` and `KszPhyPort`, maybe these ought to be renamed to `select_phy_port` and `select_port` or something?
```suggestion
    fn select_phy_port(i: KszPhyPort, r1: Register, r2: Register) -> Register {
        match i {
            KszPhyPort::One => r1,
            KszPhyPort::Two => r2,
        }
    }

    // Helper function to dispatch between three registers
    #[inline(always)]
    fn select_port(
```

Reply:
Good idea, done